File: ip-messaging/channels/result-paging/result-paging.m
// If your subsequent pages will use the same completion
// handler, you can set the block as a variable to be
// referenced from within the block.
void __block (^_completion)();
TCHChannelDescriptorPaginatorCompletion completion = ^(TCHResult *result, TCHChannelDescriptorPaginator *paginator) {
    if (result.isSuccessful) {
        NSArray<TCHChannelDescriptor *> *items = [paginator items];
        // << consume the items in your UI >>

        // You can either obtain the next page immediately or wait
        // until the user scrolls to the bottom of the list before
        // requesting, depending on your use case.
        if (paginator.hasNextPage) {
            [paginator requestNextPageWithCompletion:completion];
        }
    } else {
        // report the error to the user
    }
};
_completion = completion;
[channelsList publicChannelsWithCompletion:completion];

File: ip-messaging/result-paging/channel-descriptor-pagination/result-paging.swift

func channelsCompletion(result: TCHResult?, paginator: TCHChannelDescriptorPaginator?) {
    let items = paginator?.items()

    if result!.isSuccessful() {
        // << consume the items in your UI >>

        // You can either obtain the next page immediately or wait
        // until the user scrolls to the bottom of the list before
        // requesting, depending on your use case.
      if paginator!.hasNextPage() {
        paginator?.requestNextPage(completion: completion)
      }
    } else {
      // report the error to the user
    }
  }

client.channelsList().publicChannelDescriptors(completion: channelsCompletion)
